x/evm/keeper/grpc_query.go

```diff
@@ -672,18 +657,23 @@ func getChainID(ctx sdk.Context, chainID int64) (*big.Int, error) {
 }
 
 // same as ethTx.AsMessage, just from is provided instead of calculated from signature
-func unsignedTxAsMessage(from common.Address, ethTx *ethtypes.Transaction) ethtypes.Message {
-	return ethtypes.NewMessage(
-		from,
-		ethTx.To(),
-		ethTx.Nonce(),
-		ethTx.Value(),
-		ethTx.Gas(),
-		new(big.Int).Set(ethTx.GasPrice()),
-		new(big.Int).Set(ethTx.GasFeeCap()),
-		new(big.Int).Set(ethTx.GasTipCap()),
-		ethTx.Data(),
-		ethTx.AccessList(),
-		false, // isFake
-	)
+func unsignedTxAsMessage(from common.Address, ethTx *ethtypes.Transaction, baseFee *big.Int) *core.Message {
+	gasPrice := ethTx.GasPrice()
+	if baseFee != nil {
+		gasPrice = cmath.BigMin(ethTx.GasPrice().Add(ethTx.GasTipCap(), baseFee), ethTx.GasFeeCap())
```

Comment:
It is unclear to me why we need to add this logic here
We've already been using EIP-1559 since the last release

Reply:
I'm just trying to exactly copy the logic from the `TransactionToMessage()` `go-ethereum` function:

```
// TransactionToMessage converts a transaction into a Message.
func TransactionToMessage(tx *types.Transaction, s types.Signer, baseFee *big.Int) (*Message, error) {
	msg := &Message{
		Nonce:             tx.Nonce(),
		GasLimit:          tx.Gas(),
		GasPrice:          new(big.Int).Set(tx.GasPrice()),
		GasFeeCap:         new(big.Int).Set(tx.GasFeeCap()),
		GasTipCap:         new(big.Int).Set(tx.GasTipCap()),
		To:                tx.To(),
		Value:             tx.Value(),
		Data:              tx.Data(),
		AccessList:        tx.AccessList(),
		SkipAccountChecks: false,
		BlobHashes:        tx.BlobHashes(),
		BlobGasFeeCap:     tx.BlobGasFeeCap(),
	}
	// If baseFee provided, set gasPrice to effectiveGasPrice.
	if baseFee != nil {
		msg.GasPrice = cmath.BigMin(msg.GasPrice.Add(msg.GasTipCap, baseFee), msg.GasFeeCap)
	}
	var err error
	msg.From, err = types.Sender(s, tx)
	return msg, err
}
```